src/web/components/provider/LanguageProvider.tsx

```diff
@@ -0,0 +1,82 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import {
+  getLocale,
+  setLocale as changeGmpLocale,
+  onLanguageChange,
+} from 'gmp/locale/lang';
+import {createContext, useState, useEffect, useCallback, useMemo} from 'react';
+import useGmp from 'web/hooks/useGmp';
+
+interface LanguageContextProps {
+  language: string;
+  setLanguage: (lang: string) => Promise<void>;
+}
+
+interface LanguageProviderProps {
+  children: React.ReactNode;
+}
+
+const SETTING_ID_LOCALE = '6765549a-934e-11e3-b358-406186ea4fc5';
+
+export const LanguageContext = createContext<LanguageContextProps>({
+  language: '',
+  setLanguage: async () => {},
+});
+
+export const LanguageProvider: React.FC<LanguageProviderProps> = ({
+  children,
+}) => {
+  const gmp = useGmp();
+  const [languageState, setLanguageState] = useState<string>(
+    getLocale() ?? 'en',
```

Comment:
Nitpick, can we introduce a `const DEFAULT_LANGUAGE = 'en';` variable here (or in gmp/locale/lang)?